### PR TITLE
refactor(components): el-notification icon

### PR DESCRIPTION
**Pull request by adaex**
Please make sure these boxes are checked before submitting your PR, thank you!

- [ ] Make sure you follow contributing guide [English](https://github.com/element-plus/element-plus/blob/master/.github/CONTRIBUTING.en-US.md) | ([中文](https://github.com/element-plus/element-plus/blob/master/.github/CONTRIBUTING.zh-CN.md) | [Español](https://github.com/element-plus/element-plus/blob/master/.github/CONTRIBUTING.es.md) | [Français](https://github.com/element-plus/element-plus/blob/master/.github/CONTRIBUTING.fr-FR.md)).
- [ ] Make sure you are merging your commits to `dev` branch.
- [ ] Add some descriptions and refer to relative issues for your PR.

### CODE REVIEW

**Review by sxzz — APPROVED** [review]

**Comment by imswk** [comment]
`INotificationOptions` in `notification.type.ts` needs to be updated

**Comment by adaex** [comment]
> `INotificationOptions` in `notification.type.ts` needs to be updated

@imswk  updated 😊

File: packages/components/notification/src/index.vue [hunk 1]
```diff
@@ -100,7 +102,13 @@ export default defineComponent({
...
     const typeClass = computed(() => {
       const type = props.type
-      return type && TypeMap[type] ? `el-icon-${TypeMap[type]}` : ''
+      return type && TypeComponentsMap[props.type]
+        ? `el-notification--${type}`
+        : ''
+    })
+
+    const iconComponent = computed(() => {
+      return props.icon || TypeComponentsMap[props.type] || ''
```
Comment (imswk) [inline packages/components/notification/src/index.vue:111]: `TypeComponentsMap[props.type] `should have higher priority than` props.icon`

File: packages/components/notification/src/index.vue [hunk 2]
```diff
@@ -50,25 +51,26 @@ import { defineComponent, computed, ref, onMounted, onBeforeUnmount } from 'vue'
...
+    ElIcon,
+    ...TypeComponents,
+  },
   props: {
     customClass: { type: String, default: '' },
     dangerouslyUseHTMLString: { type: Boolean, default: false },
     duration: { type: Number, default: 4500 },
-    iconClass: { type: String, default: '' },
+    icon: {
+      type: [String, Object] as PropType<string | Component>,
```
Comment (imswk) [inline packages/components/notification/src/index.vue:71]: The `String` type is invalid and can be removed

File: docs/en-US/component/notification.md [hunk 3]
```diff
@@ -86,7 +86,7 @@ In this case you should call `ElNotification(options)`. We have also registered
 | message                  | description text                                                                                                   | string/Vue.VNode | —                                           | —         |
 | dangerouslyUseHTMLString | whether `message` is treated as HTML string                                                                        | boolean          | —                                           | false     |
 | type                     | notification type                                                                                                  | string           | success/warning/info/error                  | —         |
-| iconClass                | custom icon's class. It will be overridden by `type`                                                               | string           | —                                           | —         |
+| icon                     | custom icon component. It will be overridden by `type`                                                             | string           | —                                           | —         |
```
Comment (imswk) [inline docs/en-US/component/notification.md:89]: You can change the `string` type to `Vue.Component`

Comment (adaex) [reply docs/en-US/component/notification.md]: updated

File: packages/components/notification/src/index.vue (hunk 1 above)
Comment (adaex) [reply packages/components/notification/src/index.vue]: updated

File: packages/components/notification/src/index.vue (hunk 2 above)
Comment (adaex) [reply packages/components/notification/src/index.vue:71]: We can modify it uniformly later

Comment (imswk) [reply packages/components/notification/src/index.vue:71]: ✌️

**Pull request merged**